compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.kt

```diff
@@ -238,12 +247,20 @@ internal class MetalRedrawer(
     private fun draw(reason: DrawReason) {
         check(NSThread.isMainThread)
 
+        val caDisplayLink = caDisplayLink
+
         if (caDisplayLink == null) {
             // TODO: anomaly, log
-            // Logger.warn { "caDisplayLink callback called after it was invalidated " }
+            // Logger.warn { "draw should not be called after caDisplayLink was invalidated" }
             return
         }
 
+        // If called outside of unpaused CADisplayLink scope, targetTimestamp doesn't contain a valid value
+        val targetTimestamp = when (reason) {
+            DrawReason.DISPLAY_LINK_CALLBACK -> caDisplayLink.targetTimestamp
+            DrawReason.SYNCHRONOUS_DRAW_REQUEST -> CACurrentMediaTime()
```

Comment:
It seems that we can call rendering synchronously after we prepared the next frame.

```
animation -> render(16 * 8)
sync resize -> render(16 * 8 - 4)
vsync at 16 * 8
```
Not sure what will happen if we pass a lower time that we passed before that. The user code also will receive this time.

Should we remember the last passed time and pass the max?

Reply:
We will probably break something, fixed it.